ed25519/marshal_test.go

```diff
@@ -67,4 +60,32 @@ var _ = Describe("Surge marshalling", func() {
 			})
 		})
 	}
+	for _, t := range types {
```

Comment:
This is not quite what I was suggesting. What I was suggesting was to have this loop be over all types except for `Commitment`, and use `surgeutil.MarshalUnmarshalCheck`. Then, add an additional `Context`/`If` for just the `Commitment` case. Also, in the `else` case you are testing the `Point` type, not the `Commitment` type. I think we should test both though, so we should have two separate tests, one a manual test for `Point` and the other for `Commitment`.